doc/voq/fabric.md

```diff
@@ -155,21 +156,121 @@ When a forwarding ASIC is initialized, the fabric ports are initialized by defau
+
+## 2.8 Fabric Link Monitor
+
+SONiC needs to monitor the fabric link status and take corresponding actions once an unhealthy link is detected to avoid traffic loss. Once the fabric link monitoring feature is enabled, SONiC needs to monitor the fabric capacity of a forwarding ASIC and take corresponding action once the capacity goes below the configured threshold.
+
+The design of fabric link monitor is intentionally scoped to use local component state such as information local to a linecard or information local to a supervisor. This design simplifies the need for inter-component communication.
+
+### 2.8.1 Monitor Fabric Link Status
+
+Unhealthy fabric links may lead to traffic drops. Fabric link monitoring is an important tool to minimize traffic loss. The fabric link monitor algorithm monitors fabric link status and isolates the link if one or more criteria are true. By isolating a fabric link, the link is still up in the physical layer, but is taken out of service and does not distribute traffic. This feature is needed on both fabric ASICs and forwarding ASICs.  
+
+#### 2.8.1.1 Fabric link monitoring criteria
+
+The fabric link monitoring algorithm checks two type of errors on a link: crc errors and uncorrectable errors. 
+
+The criteria can be extended to include checking other errors later.
+
+#### 2.8.1.2 Monitoring algorithm
 
-asic2 fabric port counter (number of fabric ports: 192)
+Instead of reacting to the counter changes, Orchagent adds a new poller and periodically polls status of all fabric links. By default, the total number of received cells, cells with crc errors, cells with uncorrectable errors are fetched from all serdes links periodically and the error rates are calculated using these numbers. If any one of the error rates is above the threshold for a number of consecutive polls, the link is identified as an unhealthy link. Then the link is automatically isolated to not distribute traffic.
 
-PORT            RxCells     TxCells      Crc       Fec  Corrected
--------------------------------------------------------------------------
- 0           : 71660578         2         0         0         0
- 1           : 71659798         1         0         0       213
- 2           :        0         1         0         0       167
- 3           :        0         2         0         0       193
+#### 2.8.1.1 Cli commands
+
+Several commands will be added to set fabric link monitor config parameters.
+```
+> config fabric port monitor error threshold <#crcCells> <#rxCells>
```

Comment:
Please add Yang table schema for all new configurations proposed.

Reply:
yes, will add the Yang table for all the new configurations.